### ISSUE TITLE
Compilation benchmarks and artifacts size threshold

### ISSUE THREAD

**Issue by rzvxa** [oxc-project/backlog#86]
Now that we are leaning toward using procedural macros where appropriate, And with the addition of all the generated code we use, It would be wise to add some sort of CI step to time our compilations and prevent merge without acknowledgment if changes cause significant slow-down.

This would be hard since getting rid of variants on GitHub CI agents might prove to be a difficult problem to solve.

The other one(artifact size) is much easier to implement and has an immediate effect on how we develop, It would be awesome if we get to know how much each PR affects our build size so we can make informed decisions between performance and binary size, Everyday things like inlining and/or declarative macros are the areas which gain the most from this information.

**Comment by rzvxa** [comment]
cc @oxc-project/core

**Comment by Boshen** [comment]
I have some of it tracked https://oxc-project.github.io/monitor-oxc/metrics in https://github.com/oxc-project/monitor-oxc

**Comment by rzvxa** [comment]
Yeah, I've seen that repo before(didn't know about the webpage), However, This would be more useful to us if we could get metrics per PR.

It is similar to our oxlint ecosystem CI, It'll be fantastic if we could get these to run on related changes, Or if we don't want to slow our main CI workflow we can schedule these runs through @oxc-bot and it can report back via a comment.

#### Edit:
Something like `@oxc-bot artifacts?` and `@oxc-bot oxlint-ecosystem?` can be really useful.

**Comment by overlookmotel** [comment]
Related to #6.
